reflectapi/src/codegen/python.rs

```diff
@@ -0,0 +1,4117 @@
+
+        return Ok(result);
+    }
+
+    // Check if it's a user-defined type in the schema
+    if let Some(type_def) = schema.get_type(&type_ref.name) {
+        // Check if this is a single-field tuple struct that should be unwrapped
+        if let reflectapi_schema::Type::Struct(struct_def) = type_def {
+            if struct_def.is_tuple() && struct_def.fields.len() == 1 {
+                // Return the inner type instead of the wrapper
+                let inner_field = &struct_def.fields[0];
+                return type_ref_to_python_type(
+                    &inner_field.type_ref,
+                    schema,
+                    implemented_types,
+                    active_generics,
+                );
+            }
+        }
+
+        let base_type = improve_class_name(&type_ref.name);
+
+        // Handle generic types with arguments - follow TypeScript pattern
+        if !type_ref.arguments.is_empty() {
+            // Now that we're properly implementing generic variant classes,
+            // we can remove the inline union generation and let the regular path handle it
+
+            // Convert all arguments to Python types
+            let arg_types: Result<Vec<String>, _> = type_ref
+                .arguments
+                .iter()
+                .map(|arg| type_ref_to_python_type(arg, schema, implemented_types, active_generics))
+                .collect();
+            let arg_types = arg_types?;
+
+            // Always use bracket notation for generic types
+            // Never do string replacement in the type name itself
+            return Ok(format!("{}[{}]", base_type, arg_types.join(", ")));
+        }
+
+        return Ok(base_type);
+    }
+
+    // Try schema-based fallback first (like TypeScript codegen does)
+    if let Some(fallback_type_ref) = type_ref.fallback_once(schema.input_types()) {
+        return type_ref_to_python_type(
+            &fallback_type_ref,
+            schema,
+            implemented_types,
+            active_generics,
+        );
+    }
+
+    if let Some(fallback_type_ref) = type_ref.fallback_once(schema.output_types()) {
+        return type_ref_to_python_type(
+            &fallback_type_ref,
+            schema,
+            implemented_types,
+            active_generics,
+        );
+    }
+
+    // Final fallback for undefined external types - try to map to sensible Python types
+    let fallback_type = map_external_type_to_python(&type_ref.name);
+    Ok(fallback_type)
+}
+
+// Get type parameter names for a given type
+fn get_type_parameters(type_name: &str) -> Vec<&'static str> {
+    match type_name {
+        "std::boxed::Box" => vec!["T"],
+        "std::sync::Arc" => vec!["T"],
+        "std::rc::Rc" => vec!["T"],
+        "std::vec::Vec" => vec!["T"],
+        "std::option::Option" => vec!["T"],
+        "reflectapi::Option" => vec!["T"],
+        "std::collections::HashMap" => vec!["K", "V"],
+        "std::collections::BTreeMap" => vec!["K", "V"],
```

Comment:
This is an example where it would be better to make as "unknown to Python" type and use the fallback type provided by the BTreeMap reflected info. Basically what if somebody provides a type called `my_super_crate::ConcurrentLockFreeHashMap<K, V>` with fallback annotated as HashMap? Will your codegen still work?

Reply:
- known types with generics HashMap<String, i32> → dict[str, int]
- unknown types use the schema-level fallback resolution
- otherwise fallback to Any but preserve the original types by mapping into Python `Annotated[Any, "External type: ..."]`.


There is a very good chance that custom generic types are not handled perfectly.